### DISCUSSION TITLE
Undefined at the bottom of the pin

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#90] [Q&A]
When using the rule given below for pins with html anchors

`$text=<a href="((*@pageRef*))">((*@userstring*))</a>\n\n[img[((*@onImage*??*@imageURL*??*@largestImgURL*))]]`

![image](https://user-images.githubusercontent.com/67494083/158449325-27e4479d-b418-4819-bafe-f35a0b541275.png)

Why I get `undefined ` at the end of the pin

![image](https://user-images.githubusercontent.com/67494083/158449388-6bcc6939-d4a6-4a5d-a3ca-a64d333773c3.png)

![image](https://user-images.githubusercontent.com/67494083/158449578-233ea3cc-5f47-409a-acd1-74929b655e06.png)

But if I use the rule give below and select some  text also before pin, undefined is not seen at the end

`@text=((*@removelf(@text)*))
$text=((*$text*)) \n\n<a href="((*@pageRef*))">((*@userstring*))</a>\n\n[img[((*@onImage*??*@imageURL*??*@largestImgURL*))]]\n\n((*@text*))`

![image](https://user-images.githubusercontent.com/67494083/158449970-a110636f-641c-43a0-b7b4-424b9a91a924.png)

![image](https://user-images.githubusercontent.com/67494083/158450039-0e702768-bf18-4a53-a462-acc5890bfa18.png)

**Comment by buggyj** [comment]
can you share the table from where you reference 'pintexthtmlfieldsModifers'

**Reply by arunnbabu81** [reply]
> can you share the table from where you reference 'pintexthtmlfieldsModifers'

![image](https://user-images.githubusercontent.com/67494083/158507583-52a16287-97dd-4467-acf9-ba4176fe8ef4.png)

![image](https://user-images.githubusercontent.com/67494083/158507638-1cb6dbcc-0555-4a52-84e8-6a22f8b247d5.png)


![image](https://user-images.githubusercontent.com/67494083/158507708-ed2fc641-30ca-468d-9f84-3152f03651b7.png)

![image](https://user-images.githubusercontent.com/67494083/158507801-6ecd2ce9-421d-4ee6-a6ae-ee72d4e0a722.png)

**Reply by buggyj** [reply]
Are both cases showing  'undefined'?

**Reply by buggyj** [reply]
putting pintexthtmlfieldsModifers in the 'Body' column is probably not a good idea, better in the ''Field modifiers

**Reply by arunnbabu81** [reply]
> putting pintexthtmlfieldsModifers in the 'Body' column is probably not a good idea, better in the ''Field modifiers

This was the reason. Thank you
